Wardrobe - New Compats: CUP Units (#10905)
acemod/ACE3: docs/wiki/framework/wardrobe-framework.md

Since there is no reliable, common patterns in terms of class inheritance, not even within the same DLC, each possible variant has to be defined individually within `configFile >> "ace_wardrobe"`.

### 1.1 Components

Components are an optional assistance to define a group of variants where a variant is made of one or more components.
For example, a bandanna with sunglasses would have the the bandanna and the sunglasses as components, while the bandanna and sunglasses themself only have themselves as a component.

If the player switches from a variant with more components to a variant with less components, the surplus components will be deposited in their inventory. Vise Versa, if the player wants to change from a variant with less components to a variant with more components, they are required to have the missing components in their inventory, which will be removed on conversion.

More examples can be found below.

### 1.2 Config Guidelines

- A Container (Uniform, Vest, Backpack) should never change its `maxLoad`, unless there is a really good reason for it. This will result in the loss of items.

> - Variants should not "magically" add/remove parts to/from themselves, unless its handled through a component, especially when they add protection or functionality.

Reviewer: ```suggestion
- Variants should not "magically" add/remove parts to/from themselves, unless it's handled through a component, especially when they add protection or functionality.
```

>   - Do: Uniforms with and without Gloves should be fine in most cases, as they are cosmetical and dont break continuity/logic.

Reviewer: ```suggestion
  - Do: Uniforms with and without Gloves should be fine in most cases, as they are cosmetic and don't break continuity/logic.
```

  - Do: Helmets with integral flip-down visors.
  - Don't: Helmet with additional armor or face protector without the requirement of some component that will be removed from the inventory.

> - There should not be a difference in weight or container capacity between the different variants unless compoents get added/removed.

Reviewer: ```suggestion
- There should not be a difference in weight or container capacity between the different variants unless components get added/removed.
```

    class G_Bandanna_blk: ace_wardrobe_base {
        class modifiableTo {
            class G_Bandanna_aviator;
            class G_Aviator;
        };
        components[] = {"G_Bandanna_blk"};
    };

    class G_Aviator: ace_wardrobe_base {
        class modifiableTo {
            class G_Bandanna_aviator;
        };

        components[] = { "G_Aviator" };
    };

    class G_Bandanna_aviator: ace_wardrobe_base {
        class modifiableTo {
            class G_Bandanna_blk;
            class G_Aviator;
        };
        components[] = { "G_Aviator", "G_Bandanna_blk" };
    };
};
```

## 5. Sounds
The following `CfgSounds` classes are integrated in `ace_wardrobe` and are the default sounds for the `ace_wardrobe_base` and `ace_wardrobe_base_H_visor_up`/`ace_wardrobe_base_H_visor_down` base classes.

### 5.1 Integrated Sounds
The number at the end of the classnames indicates the length of the file in 1/10th seconds.
10 -> 1 sec, 15 -> 1.5sec, ...

- `ace_wardrobe_fabric_06`
- `ace_wardrobe_fabric_07`
- `ace_wardrobe_fabric_16`
- `ace_wardrobe_fabric_20`
- `ace_wardrobe_fabric_25`
- `ace_wardrobe_helmet_visor_05`

## 6. Compatibility

## 6.1 MagzineID

> Currently, `ace_IntelItems` and `ace_overheating` (spare barrels) are being directly supported.

Reviewer: ```suggestion
Currently, `ace_intelitems` and `ace_overheating` (spare barrels) are being directly supported.
```

> If an addon or mod utilizes a magazine's `magazineID` to handle additional data about items carried by the player, then the process of modifying a wearable container (uniform, vest, backpack) to another variant will result in new `magazineID`s for all magazines on the player and therefore, require special handling within ace_wardrobe functions.

Reviewer: ```suggestion
If an addon or mod utilizes a magazine's `magazineID` to handle additional data about items carried by the player, then the process of modifying a wearable container (uniform, vest, backpack) to another variant will result in new `magazineID`s for all magazines on the player and therefore, require special handling within `ace_wardrobe` functions.
```

There *might* be some changes coming to how arma handles the `setUnitLoadout` which could resolve the need for this.

## 6.2 Container Size - Uniform, Vest, Backpack
When the player changes from one container item to another through the wardrobe action and the container's `maximumLoad` is smaller then previously, the player risks the loss of items carried inside said container.

> Therefore, the function `[] call ace_wardrobe_fnc_compare_container_maxLoad` can be used to compare the item's maximumLoad. The result will be dumped into the .rpt.

Reviewer: ```suggestion
Therefore, the function `[] call ace_wardrobe_fnc_compare_container_maxLoad` can be used to compare the item's `maximumLoad`. The result will be dumped into the .rpt.
```
